### ISSUE TITLE
missing dependency spec or version boost?  ModuleNotFoundError: No module named 'wx.siplib'

### ISSUE THREAD

**Issue by yarikoptic** [pauldmccarthy/fsleyes#29]
trying to run fsleyes on my debian laptop, while relying on system wide installed libraries and having disabled demanding PyOpenGL-accelerate (#27):

<details>
<summary>here is the full protocol of installation</summary> 

```shell
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> virtualenv --python python3 --system-site-packages venvs/dev3
Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3/bin/python3
Also creating executable in /home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
1 20783.....................................:Fri 16 Aug 2019 12:01:58 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> source venvs/dev3/bin/activate                               
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20784.....................................:Fri 16 Aug 2019 12:02:04 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> git diff
diff --git a/requirements.txt b/requirements.txt
index 2c98ada9..2c348fa4 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -6,7 +6,7 @@ nibabel>=2.3.*
 Jinja2==2.*
 Pillow>=3.2.0
 PyOpenGL>=3.1.0
-PyOpenGL-accelerate>=3.1.0
+#PyOpenGL-accelerate>=3.1.0
 pyparsing==2.*
 scipy>=0.18
 wxPython>=3.0.2.0
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20785.....................................:Fri 16 Aug 2019 12:02:07 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> pip install -e .                                             
Obtaining file:///home/yoh/deb/gits/pkg-exppsy/fsleyes
Requirement already satisfied: setuptools in ./venvs/dev3/lib/python3.7/site-packages (from fsleyes==0.30.1) (41.1.0)
Requirement already satisfied: six==1.* in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (1.12.0)
Requirement already satisfied: numpy>=1.14 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (1.16.2)
Requirement already satisfied: matplotlib>=1.5.1 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (3.0.2)
Requirement already satisfied: nibabel>=2.3.* in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (2.5.0)
Requirement already satisfied: Jinja2==2.* in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (2.10)
Requirement already satisfied: Pillow>=3.2.0 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (5.4.1)
Requirement already satisfied: PyOpenGL>=3.1.0 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (3.1.0)
Requirement already satisfied: pyparsing==2.* in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (2.2.0)
Requirement already satisfied: scipy>=0.18 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (1.1.0)
Requirement already satisfied: wxPython>=3.0.2.0 in /usr/lib/python3/dist-packages (from fsleyes==0.30.1) (4.0.4)
Collecting fslpy>=2.4 (from fsleyes==0.30.1)
  Using cached https://files.pythonhosted.org/packages/bb/ed/83f9b3c10d6063393de88d3a083ef11996eacf7d32c7080b6f3e723e885e/fslpy-02.5.0-py2.py3-none-any.whl
Collecting fsleyes-props>=1.6.2 (from fsleyes==0.30.1)
  Using cached https://files.pythonhosted.org/packages/37/35/ee31cc0a17f3f209d65d360257ceaf27469fcbe7be7efc6f05742bfed5a5/fsleyes_props-1.6.5-py2.py3-none-any.whl
Collecting fsleyes-widgets>=0.6 (from fsleyes==0.30.1)
  Using cached https://files.pythonhosted.org/packages/33/ab/0bd722e5b382f86252caddab530765df1316944f329054fc66169044925b/fsleyes_widgets-0.7.3-py2.py3-none-any.whl
Requirement already satisfied: h5py>=2.9 in /usr/lib/python3/dist-packages (from fslpy>=2.4->fsleyes==0.30.1) (2.9.0)
Installing collected packages: fslpy, fsleyes-widgets, fsleyes-props, fsleyes
  Running setup.py develop for fsleyes
Successfully installed fsleyes fsleyes-props-1.6.5 fsleyes-widgets-0.7.3 fslpy-2.5.0
pip install -e .  6.20s user 0.97s system 98% cpu 7.300 total
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20786.....................................:Fri 16 Aug 2019 12:02:16 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> which fsleyes
/usr/bin/fsleyes
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20787.....................................:Fri 16 Aug 2019 12:02:23 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> rehash
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20788.....................................:Fri 16 Aug 2019 12:02:25 PM EDT:.
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> which fsleyes
/home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3/bin/fsleyes
changes on filesystem:                                                                                                                                                                                                            
 requirements.txt | 2 +-
(dev3) 1 20789.....................................:Fri 16 Aug 2019 12:02:26 PM EDT:.

```
</details>

but running fsleyes fails with
```
(git)hopa:~exppsy/fsleyes[tags/0.30.1]git
$> fsleyes
Traceback (most recent call last):
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3/bin/fsleyes", line 11, in <module>
    load_entry_point('fsleyes', 'gui_scripts', 'fsleyes')()
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/filtermain.py", line 169, in main
    result = fm.main(args)
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/main.py", line 213, in main
    fsleyes.initialise()
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/__init__.py", line 253, in initialise
    import fsleyes.plugins as plugins
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/plugins.py", line 177, in <module>
    import fsleyes.views.viewpanel       as viewpanel
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/views/viewpanel.py", line 21, in <module>
    import fsleyes.panel                 as fslpanel
  File "/home/yoh/deb/gits/pkg-exppsy/fsleyes/fsleyes/panel.py", line 379, in <module>
    import wx.siplib as sip
ModuleNotFoundError: No module named 'wx.siplib'
```

<details>
<summary>
and unfortunately installation fails if attempted without relying on system wide installed modules since seems wxPython missing requirement for pathlib2
</summary> 

```
  ERROR: Command errored out with exit status 1:
   command: /home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3-standalone/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/yoh/.tmp/pip-install-q4oe0iq7/wxPython/setup.py'"'"'; __file__='"'"'/home/yoh/.tmp/pip-install-q4oe0iq7/wxPython/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /home/yoh/.tmp/pip-wheel-jdvje8sx --python-tag cp37
       cwd: /home/yoh/.tmp/pip-install-q4oe0iq7/wxPython/
  Complete output (14 lines):
  running bdist_wheel
  running build
  WARNING: Building this way assumes that all generated files have been
  generated already.  If that is not the case then use build.py directly
  to generate the source and perform the build stage.  You can use
  --skip-build with the bdist_* or install commands to avoid this
  message and the wxWidgets and Phoenix build steps in the future.
  
  "/home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3-standalone/bin/python3" -u build.py build
  Traceback (most recent call last):
    File "build.py", line 30, in <module>
      import pathlib2
  ModuleNotFoundError: No module named 'pathlib2'
  Command '"/home/yoh/deb/gits/pkg-exppsy/fsleyes/venvs/dev3-standalone/bin/python3" -u build.py build' failed with exit code 1.
  ----------------------------------------
  ERROR: Failed building wheel for wxPython
```
</details>
preinstalling that one leads to other rabbit holes, ... might look more into it later

**Comment by pauldmccarthy** [comment]
Hmm, I'm not sure about the `wx.siplib` issue - it looks like the debian version of `wxpython` is missing this package.

Installing wxPython via pypi does work, but the catch is that Linux binaries are not currently hosted on pypi (I think they're too big or something).  So you need to install directly from [here](https://extras.wxpython.org/wxPython4/extras/linux/), using something like:

```
pip install -f https://extras.wxpython.org/wxPython4/extras/linux/gtk2/debian-9/ wxpython
```

(I haven't yet made the leap to `gtk3` - not sure what the FSLeyes UI is like if you go with that)

**Comment by yarikoptic** [comment]
FWIW
- `apt-get install libgtk-3-dev` and that allowed for wxpython to finally build (also added pathlib2 into requirements.txt here)
- either installations or running this fresh new fsleyes "fixed" the issue of system wide (old 0.15.2 from neurodebian) segfaulting upon start
- indeed there is 
```
$> find -iname siplib*
./venvs/dev3-standalone/lib/python3.7/site-packages/wx/siplib.cpython-37m-x86_64-linux-gnu.so
```
which is not provided by wx on debian which is `python3-wxgtk4.0` so -- for gtk4
- in Debian pkg it was explicitly "removed" from the build: https://salsa.debian.org/freewx-team/wxpython4.0/blob/master/debian/patches/unbundle-sip.patch  I filed an issue and CCed you.

Cheers
